libafl/src/corpus/cached.rs

```diff
@@ -20,27 +20,13 @@ use crate::{
 /// The eviction policy is FIFO.
 #[cfg(feature = "std")]
 #[derive(Default, Serialize, Deserialize, Clone, Debug)]
-#[serde(bound = "I: serde::de::DeserializeOwned")]
-pub struct CachedOnDiskCorpus<I>
-where
-    I: Input,
-{
+pub struct CachedOnDiskCorpus<I> {
```

Comment:
The one thing I don't fully understand: `CachedOnDiskCorpus` is never supposed to be used with something that's not an `Input`, right? So why wouldn't we want to specify the constraint here?

Reply:
By leaving it unconstrained, we effectively make it so users can say "don't care". If you don't use it as an input, then you don't need to specify it as such. This allows us to remove a ton of generic bounds downstream.